primitives/runtime/src/messages.rs

```diff
@@ -22,7 +22,7 @@ use scale_info::TypeInfo;
 
 /// Message dispatch result.
 #[derive(Encode, Decode, RuntimeDebug, Clone, PartialEq, Eq, TypeInfo)]
-pub struct MessageDispatchResult {
+pub struct MessageDispatchResult<DispatchLevelResult: Clone + Decode + sp_std::fmt::Debug + Eq> {
```

Comment:
Can you also try `NoBound` versions here? :) I've found them recently && iiuc it should help with removing these constraints (maybe I'm wrong though)

Reply:
when I tried to change `RuntimeDebug` to `RuntimeDebugNoBound` I get this error:
![image](https://user-images.githubusercontent.com/8159517/204839511-c344491d-4624-4d9c-9d17-7489b1bdaeef.png)